project/build.scala

```diff
@@ -20,20 +21,32 @@ object BuildSettings extends Build {
   )
 
   lazy val chisel = project in file("chisel3")
-  lazy val hardfloat  = project.dependsOn(chisel)
-  lazy val macros = project in file("macros")
-  lazy val rocketchip = (project in file(".")).settings(chipSettings).dependsOn(chisel, hardfloat, macros)
+  lazy val hardfloat  = project.dependsOn(chisel).aggregate(chisel)
+  lazy val macros = Project(
+    id   = "rocket-macros",
+    base = file("macros")
+  )
+  lazy val rocketchip = (project in file(".")).settings(chipSettings).dependsOn(chisel, hardfloat, macros).aggregate(chisel, hardfloat, macros)
 
   lazy val addons = settingKey[Seq[String]]("list of addons used for this build")
   lazy val make = inputKey[Unit]("trigger backend-specific makefile command")
   val setMake = NotSpace ~ ( Space ~> NotSpace )
 
+  val resourceSubdirs = Seq("vsrc", "csrc", "bootrom", "src/main/resources")
```

Comment:
My previous commit used symlinks, but instead I'm playing around with letting sbt know where the resources are.

One limitation of this is that any resources under src/main/resources show up under src/main/resources inside the jar. It doesn't seem like sbt gives a great way to deal with this.

Reply:
According to the [sbt documentatio](http://www.scala-sbt.org/1.x/docs/Howto-Customizing-Paths.html)n, you should be able to use something like:
```scala
unmanagedResourceDirectories in Compile += baseDirectory.value / "extra-resources"
```